vm-setup/roles/v1aX_integration_test/tasks/main.yml

```diff
@@ -1,78 +1,89 @@
 ---
+
+- name: Install openshift client in CentOS 8 using pip3
+  pip:
+    executable: "pip3"
+    name: openshift
+    state: present
+  become: true
+  become_user: root
+  when: ansible_facts['distribution'] == "CentOS" and ansible_facts['distribution_major_version'] == '8'
+
+- name: Install openshift client in CentOS 7 using pip
+  pip:
+    executable: "pip"
+    name: openshift
+    state: present
+  become: true
+  become_user: root
+  when: ansible_facts['distribution'] == "CentOS" and ansible_facts['distribution_major_version'] == '7'
+
 - name: "Generate templates for v1alpha3"
   include: "{{ item }}.yml"
   with_items:
     - generate_templates
 
-- name: "{{ CAPI_VERSION }} deployment"
-  include: "{{ item }}.yml"
-  with_items:
-    - provision_cluster
-    - download_image
-    - provision_controlplane
-    - provision_worker
-    - verify
-    - deprovision_worker
-    - deprovision_controlplane
-    - deprovision_cluster
-    - cleanup
-  when: v1aX_integration_test_action == "ci_test"
+- name: download image for deployment
+  include: download_image.yml
+  when: v1aX_integration_test_action in image_download_actions
 
-- name: "{{ CAPI_VERSION }} cluster provisioning"
-  include: "provision_cluster.yml"
-  when: v1aX_integration_test_action == "provision_cluster"
+- name: Provision cluster
+  k8s:
+    state: present
+    src: "{{ TEMP_GEN_DIR }}/{{ CAPI_VERSION }}_cluster_{{ IMAGE_OS|lower }}.yaml"
+    namespace: "{{ NAMESPACE }}"
+  when: v1aX_integration_test_action in provision_cluster_actions
 
-- name: "{{ CAPI_VERSION }} controlplane node provisioning"
-  include: "{{ item }}.yml"
-  with_items:
-    - download_image
-    - provision_controlplane
-  when: v1aX_integration_test_action == "provision_controlplane"
+- name: Create control plane
+  k8s:
+    state: present
+    src: "{{ TEMP_GEN_DIR }}/{{ CAPI_VERSION }}_controlplane_{{ IMAGE_OS|lower }}.yaml"
+    namespace: "{{ NAMESPACE }}"
+  when: v1aX_integration_test_action in provision_controlplane_actions
 
-- name: "{{ CAPI_VERSION }} worker node provisioning"
-  include: "{{ item }}.yml"
-  with_items:
-    - download_image
-    - provision_worker
-  when: v1aX_integration_test_action == "provision_worker"
+- name: Create worker nodes
+  k8s:
+    state: present
+    src: "{{ TEMP_GEN_DIR }}/{{ CAPI_VERSION }}_workers_{{ IMAGE_OS|lower }}.yaml"
+    namespace: "{{ NAMESPACE }}"
+    when: v1aX_integration_test_action in provision_workers_actions
```

Comment:
Ops, there is a white space before the condition.

Reply:
typo alert! fixed, thank you